File: frontend/src/app/core/use-case/get-accident-reports.use-case.ts
import { Injectable, inject } from '@angular/core';
import { GetAccidentReportsPort } from '../port/out/accident.port';
import type { AccidentReport, AccidentReportSearch } from '../domain/accident.model';
import { Observable } from 'rxjs';

/**
 * Caso de uso: buscar accidentes por provincia y rango de fechas (informes).
 * Depende del puerto de salida (adaptador HTTP en runtime).
 */
@Injectable({ providedIn: 'root' })
export class GetAccidentReportsUseCase {
  private readonly port = inject(GetAccidentReportsPort);

  searchReports(params: AccidentReportSearch): Observable<AccidentReport[]> {
    return this.port.searchReports(params);
  }
}

File: backend/src/main/java/com/joaquin/proyecto_backend/application/port/in/GetAccidentReportsUseCase.java
package com.joaquin.proyecto_backend.application.port.in;

import com.joaquin.proyecto_backend.domain.AccidentReport;

import java.time.LocalDate;
import java.util.List;

/**
 * Puerto de entrada (caso de uso): listar accidentes por provincia y rango de fechas.
 */
public interface GetAccidentReportsUseCase {

    List<AccidentReport> getByProvinceAndDateRange(String province, LocalDate from, LocalDate to);
}
